dist/tools/testrunner/testrunner.py

```diff
@@ -10,14 +11,28 @@
 import sys
 import subprocess
 import time
-from traceback import print_tb
+from traceback import extract_tb, print_tb
 
-from pexpect import spawnu, TIMEOUT
+import pexpect
 
+PEXPECT_PATH = os.path.dirname(pexpect.__file__)
+RIOTBASE = os.environ['RIOTBASE'] or \
+           os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", ".."))
+
+def find_exc_origin(exc_info):
+    pos = list_until(extract_tb(exc_info),
+                     lambda frame: frame.filename.startswith(PEXPECT_PATH)
+                    )[-1]
+    return pos.line, \
+           os.path.relpath(os.path.abspath(pos.filename), RIOTBASE), \
+           pos.lineno
```

Comment:
I think its weird that this function is above the function it is using (`list_until()`) but maybe that is just personal taste

Reply:
addressed your comment and squashed immediately @miri64